pkg/authorization/toplevel_org_authorizer.go

```diff
@@ -121,44 +159,56 @@ func (a *topLevelOrgAccessAuthorizer) Authorize(ctx context.Context, attr author
 				continue
 			}
 			if subjectTopLevelOrg == requestTopLevelOrgName {
+				kaudit.AddAuditAnnotations(
+					ctx,
+					TopLevelContentAuditDecision, DecisionAllowed,
+					TopLevelContentAuditReason, "serviceaccount belongs to this top level workspace hierarchy",
+				)
 				return a.delegate.Authorize(ctx, attr)
 			}
 		}
 
+		kaudit.AddAuditAnnotations(
+			ctx,
+			TopLevelContentAuditDecision, DecisionNoOpinion,
+			TopLevelContentAuditReason, "serviceaccount does not belong to this top level workspace hierarchy",
+		)
 		return authorizer.DecisionNoOpinion, WorkspaceAcccessNotPermittedReason, nil
-
 	case isUser:
-		var (
-			errList    []error
-			reasonList []string
-		)
-
-		for _, verb := range []string{"access"} {
-			workspaceAttr := authorizer.AttributesRecord{
-				User:        attr.GetUser(),
-				Verb:        verb,
-				APIGroup:    tenancyv1alpha1.SchemeGroupVersion.Group,
-				APIVersion:  tenancyv1alpha1.SchemeGroupVersion.Version,
-				Resource:    "workspaces",
-				Subresource: "content",
-				Name:        requestTopLevelOrgName,
-
-				ResourceRequest: true,
-			}
+		workspaceAttr := authorizer.AttributesRecord{
+			User:        attr.GetUser(),
+			Verb:        "access",
+			APIGroup:    tenancyv1alpha1.SchemeGroupVersion.Group,
+			APIVersion:  tenancyv1alpha1.SchemeGroupVersion.Version,
+			Resource:    "workspaces",
+			Subresource: "content",
+			Name:        requestTopLevelOrgName,
+
+			ResourceRequest: true,
+		}
 
-			dec, reason, err := a.rootAuthorizer.Authorize(ctx, workspaceAttr)
-			if err != nil {
-				errList = append(errList, err)
-				reasonList = append(reasonList, reason)
-				continue
-			}
-			if dec == authorizer.DecisionAllow {
-				return a.delegate.Authorize(ctx, attr)
-			}
+		dec, reason, err := a.rootAuthorizer.Authorize(ctx, workspaceAttr)
+		if err != nil {
+			kaudit.AddAuditAnnotations(
+				ctx,
+				TopLevelContentAuditDecision, DecisionNoOpinion,
+				TopLevelContentAuditReason, fmt.Sprintf("%s (%v)", reason, err))
```

Comment:
as per https://github.com/kcp-dev/kcp/issues/1307, let's use `TopLevelContentAuditReason, fmt.Sprintf("forbidden by \"content\" verb RBAC in top level workspace %q, reason=%q, error=%v", requestTopLevelOrgName, reason, err))`